proto/proto_conn.h

```diff
@@ -17,9 +17,16 @@ struct sock_addr;
  * only if ${nokeepalive} is zero.  Drop the connection if the handshake or
  * connecting to the target takes more than ${timeo} seconds.  When the
  * connection is dropped, invoke ${callback_dead}(${cookie}).  Free ${sas}
- * once it is no longer needed.
+ * once it is no longer needed.  Return a cookie which can be passed to
+ * proto_conn_drop.
  */
-int proto_conn_create(int, struct sock_addr **, int, int, int, int,
-    const struct proto_secret *, double, int (*)(void *), void *);
+void * proto_conn_create(int, struct sock_addr **, int, int, int, int,
+    const struct proto_secret *, double, int (*)(void *, void *), void *);
+
+/** proto_conn_drop(conn_cookie):
```

Comment:
wow, it took me 2 whole minutes to spot the problem, even after you'd isolated the line!  (I actually thought you were complaining about `proto_conn_create`, but couldn't spot any problems there, either)